include/tvm/runtime/container.h

```diff
@@ -24,6 +24,19 @@
 #ifndef TVM_RUNTIME_CONTAINER_H_
 #define TVM_RUNTIME_CONTAINER_H_
 
+#if defined(__cpp_lib_experimental_string_view) && \
+    __cpp_lib_experimental_string_view >= 201411
+#define TVM_USE_CXX14_STRING_VIEW
+#elif defined(__cpp_lib_string_view) && __cpp_lib_string_view >= 201606
+#define TVM_USE_CXX17_STRING_VIEW
```

Comment:
We shall prioritize CXX17 string view as it is the std. also when if we defint it this way, it can leave certain macro undefined. I think we should still define it in the old way as given in the example so that the macro is always defined

Reply:
Good point! I have improved it to ensure both macros are defined. The old way will cause build warning `Macro expansion producing 'defined' has undefined behavior`,